Fhi.HelseId.Web/Services/HelseIdSecretHandlers.cs

```diff
@@ -305,38 +243,23 @@ public InvalidEnterpriseCertificateSecretException(string secret) : base(Standar
             public string Secret { get; }
         }
 
-        protected override JsonWebKey GetJsonWebKey()
-        {
-            return JsonWebKey;
-        }
-    }
-
-    public class HelseIdSharedSecretHandler : SecretHandlerBase
-    {
-        public HelseIdSharedSecretHandler(IHelseIdWebKonfigurasjon helseIdWebKonfigurasjon) : base(helseIdWebKonfigurasjon)
-        {
-        }
-
-        public override void AddSecretConfiguration(OpenIdConnectOptions options)
-        {
-            options.ClientSecret = ConfigAuth.ClientSecret;
-        }
-
-        protected override JsonWebKey GetJsonWebKey()
-        {
-            return JsonWebKey;
-        }
+        protected override SecurityKey GetSecurityKey() => _x509SecurityKey;
     }
 
     public class HelseIdNoAuthorizationSecretHandler : SecretHandlerBase
     {
+
         public HelseIdNoAuthorizationSecretHandler(IHelseIdWebKonfigurasjon helseIdWebKonfigurasjon) : base(helseIdWebKonfigurasjon)
         {
         }
 
-        protected override JsonWebKey GetJsonWebKey()
+        /// <summary>
+        /// Will never be called
+        /// </summary>
+        /// <returns></returns>
+        protected override SecurityKey GetSecurityKey()
         {
-            return JsonWebKey;
+            throw new NotImplementedException("This method is not used and should not be called. If this happens, there is a missing check for AuthUse");
```

Comment:
I find this message a little confusing. Suggestion: "This method should not be called in combination with HelseIdNoAuthorizationSecretHandler. HelseIdNoAuthorizationSecretHandler does not require a securitykey."

Reply:
If you get this exception, it is because you have enabled Auth AND set up HelseIdNoAuthorizationSecretHandler  as you security key handler.  The message attempts to explain that.